pkg/auth/auth.go

```diff
@@ -51,51 +57,102 @@ func decode64(s string) (string, error) {
 	return string(b), err
 }
 
-// Extract takes the User from middleware.
-// It always returns a User
-func Extract(ctx context.Context) *User {
-	// check if User is in go Context
+// ReadUserFromContextOrDie returns a user from the ctx or an error if none was found, or it is invalid
+func ReadUserFromContextOrDie(ctx context.Context) (*User, error) {
```

Comment:
`OrDie` usually implies that the function panics but this one does not.

Reply:
i renamed the functions